### DISCUSSION TITLE
DataPortal Exception in an existing application

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#4217] [Questions] [answered]
We are maintaining an older Csla application and we are getting 

> Csla.DataPortalException: 'DataPortal.Create failed (An item with the same key has already been added.)

What could be the possible reason?
Thanks

**Comment by Chicagoan2016** [comment]
We just restored older code from source control : )

**Reply by rockfordlhotka** [reply]
Can we get the stack trace? And the version of CSLA you are using?

**Reply by Chicagoan2016** [reply, marked as answer]
We had the same exception today, turned out we were calling 
InputProperties.Add(...) twice for the same property
